### PR TITLE
fix(js_semantic): add TS_QUALIFIED type for reference

### PR DESCRIPTION
**Pull request by fireairforce**
## Summary

Fix discussion: https://github.com/biomejs/biome/discussions/2059

### CODE REVIEW

File: crates/biome_js_semantic/src/events.rs [hunk 1]
```diff
@@ -795,6 +812,16 @@ impl SemanticEventExtractor {
                                 }
                             }
                         }
+                        Reference::Read(range) => {
```
[the remark this answers is not in the document]
Comment (Conaclos) [reply crates/biome_js_semantic/src/events.rs:815]: Another idea: We could add a new variant to `Reference`: `Reference::Qualified`. If we skipped [at least one qualified name](https://github.com/biomejs/biome/blob/main/crates/biome_js_semantic/src/events.rs#L555) and we end in the [match branch `Some(TS_REFERENCE_TYPE | TS_NAME_WITH_TYPE_ARGUMENTS)`](https://github.com/biomejs/biome/blob/main/crates/biome_js_semantic/src/events.rs#L559), then we could emit a `BindingName::Value` with a `Reference::Qualified`. Emitting a value should allow the default path to bind the ref to a variable (in our case a namespace). Similarly to `Refrence::TypeOf`, we could handle in `pop_scope` the specific case where a namespace is imported a type.

[the remark this answers is not in the document]
Comment (fireairforce) [reply crates/biome_js_semantic/src/events.rs]: ok, i have done this~

**Comment by fireairforce** [comment]
image

I am not sure if this failed for my pr..

**Review by ematipico — APPROVED** [review]
Looks good to me. cc @Conaclos 

The CI failure isn't your fault

**Review by Conaclos — COMMENTED** [review]
LGTM, the only missing change is adding `| Reference::Qualified(range)` at [this line](https://github.com/biomejs/biome/pull/2092/files#diff-0d4615b2d28fab5564365eb041ff7065f0e2dcbd7b95d2bb57c6cb3a7bfbbe31R805).

This will handle case like the following:

```ts
import { type A } from "./a.ts";
let x: A.B;
```

**Review by Conaclos — APPROVED** [review]

**Comment by Conaclos** [comment]
Thanks!

**Pull request merged**